design/specs/foreign-key-constraints.md

```diff
@@ -2,75 +2,209 @@
+  - The column contains the referenced values for each cell.
+  - When clicked, a pre-filtered list is displayed inside the [record selector](#record-selector) component. The dropdown includes a search input, that if modified, will show other matching records. Clicking on any of those records will close the dropdown and replace the existing one.
+  - Note that in the case of multi-column foreign key constraints, the system could modify additional fields.
+
+### Scenario 5b: The field is empty
+
+#### Steps for 5b
+
+- The user opens a table that has columns with a foreign key constraint applied.
+  - The column contains no values.
+  - When clicked, a list of records is displayed inside the [record selector](#record-selector) component. The list shows the first 50 records.
+  - An text input field is available within the dropdown to filter the records based on partial or complete value matches. As the user types in a value, the list narrows down the records. The system can search only the first five fields. Search across all types is supported.
+  - If there's a single match for the entered value, it should become highlighted, allowing the user to confirm the selection.
+  - If the table contains no records, the user sees an option to add a record.
+  - If there aren't any matches, the user sees an option to add a record for the newly entered value or try changing the 'Search Columns' table preference.
+
+## Scenario 6: User changes the table link preferences
+
+### Scenario 6a: From the column menu
+
+- The user opens the column menu for a column that has a foreign key constraint applied
+- The user clicks on the 'Table Link Preferences' option
+
+### Scenario 6b: From the record selector menu
+
+- The user opens the column menu for a column that has a foreign key constraint applied
+- The user clicks on the preferences link
+
+## Components
+
+### Record Selector
+
+The record selector component is used to retrieve records from other tables and add them as values to cells according to the referenced column.
+
+### Table Selector
+
+The table selector allows users to select tables from a schema. This spec doesn't go into details on how this selector would work. It also doesn't consider the scenario for selecting both tables and views.
+
+### Record Preview
+
+The record preview allows users to identify records linked in other tables. It shows the first five field names and values. A specific limit for the value length needs to be defined so that space usage is optimal.
+
+## Other Considerations
+
+### Linked Records for Multi-Column Foreign Key Constraints
+
+- The current UI does not consider multi-column foreign key constraints and how those would be retrieved and selected.
+
+### Usage of Color
+
+- A suggestion to use color as a means to differentiate table references has been included in this spec. However, there are implementation details that need to be discussed before the team can make a decision.
```

Comment:
Decision about what?

Reply:
@kgodey I've updated to add more detail.